### PR TITLE
Product import tweaks

### PR DESCRIPTION
**Pull request by snopoke**
http://manage.dimagi.com/default.asp?173239

@esoergel 

@TylerSheffels buddy

### CODE REVIEW

**Comment by snopoke** [comment]
@esoergel I wasn't sure if I should also try and validate the uniqueness of product code as well.

**Comment by TylerSheffels** [comment]
:+1: from me, but will let Ethan merge

**Comment by czue** [comment]
@snopoke right now the importer just happily lets you import duplicate codes? that feels like another bug. merging this now but can you also file a new interrupt ticket for that if what I said is true?

File: corehq/apps/products/bulk.py [hunk 1]
```diff
@@ -51,6 +54,15 @@ def import_products(domain, importer):
...
             seen_product_ids.add(p.code)
 
+        if p.program_id and p.program_id not in program_ids:
+            results['errors'].append(_(
+                u"Product {product_name} references a program that doesn't exist: {program_id}"
+            ).format(
+                product_name=p.name,
+                program_id=p.program_id
+            ))
+            continue
```
Comment (esoergel) [inline corehq/apps/products/bulk.py]: :+1:

**Comment by esoergel** [comment]
Looks good to me.  It looks like `code` should be unique per domain and always lowercase (the `get_by_code` method uses `db.view(...).first()`), although I don't think that's enforced at the db level.  We should add that to the import validation.

**Comment by snopoke** [comment]
@esoergel see https://github.com/dimagi/commcare-hq/pull/7276

**Pull request merged**